Content.Server/RandomMetadata/RandomMetadataSystem.cs

```diff
@@ -21,16 +22,28 @@ private void OnMapInit(EntityUid uid, RandomMetadataComponent component, MapInit
     {
         var meta = MetaData(uid);
 
-        if (component.NameSet != null)
+        if (component.NameSegments != null)
         {
-            var nameProto = _prototype.Index<DatasetPrototype>(component.NameSet);
-            meta.EntityName = _random.Pick(nameProto.Values);
+            var outputSegments = new List<string>();
+            foreach (var segment in component.NameSegments)
+            {
+                outputSegments.Add(_prototype.TryIndex<DatasetPrototype>(segment, out var proto)
+                    ? _random.Pick(proto.Values)
+                    : segment);
+            }
+            meta.EntityName = string.Join(" ", outputSegments);
```

Comment:
you should probably be able to specify the separator as a datafield too but eh

Reply:
yeah, should probably do this, e.g., lizards with [Noun]-the-[Verb], but this is a poor example given that randomization of lizard names is already handled within the character profile randomization system